### PR TITLE
[UN-1507] Merge getWeb3Provider and checkIfWalletInstalled

### PR DESCRIPTION
**Pull request by weeebr**
https://popstand.atlassian.net/browse/UN-1507

bonus: 
- [x] more improvements for `wallet/index.ts`

### CODE REVIEW

File: src/wallet/index.ts [hunk 1]
```diff
@@ -300,31 +294,22 @@ export const getWeb3Provider = async (): Promise<Web3Provider> => {
...
-  return ethereum;
+  return window?.ethereum?.isMetaMask || false;
 };
 
 /**
  * Returns connected wallet account (Attempts to connect to the wallet if not connected)
  * @returns Account address
  */
-export const getWalletAccount = async () => {
+export const getCurrentAddress = async () => {
```
Comment (tgmarinho) [inline src/wallet/index.ts:312]: getCurrentAddress -> getCurrentWalletAddress ?

File: src/wallet/index.ts [hunk 2]
```diff
@@ -300,31 +294,22 @@ export const getWeb3Provider = async (): Promise<Web3Provider> => {
 /**
  * Check if a web3 wallet is installed
  *
- * @returns installed web3 provider or null if none
+ * @returns boolean
```
Comment (tgmarinho) [inline src/wallet/index.ts:297]: @returns boolean | throw an Error ?

File: src/wallet/index.ts [hunk 3]
```diff
@@ -300,31 +294,22 @@ export const getWeb3Provider = async (): Promise<Web3Provider> => {
...
-
-  try {
-    // is there a more agnostic way to check? I know otherwallets use isMetaMask too, but perhaps there are better flags
-    const check = ethereum.isMetaMask;
-  } catch (e) {
-    return null;
-  }
 
-  return ethereum;
+  return window?.ethereum?.isMetaMask || false;
```
Comment (tgmarinho) [inline src/wallet/index.ts:305]:
```
return !!window?.ethereum?.isMetaMask;
```


Screen Shot 2023-02-09 at 09 55 33

**Review by tgmarinho — CHANGES_REQUESTED** [review]

File: src/wallet/index.ts (hunk 2 above)
Comment (weeebr) [reply src/wallet/index.ts:297]: no, if throw, no return

File: src/wallet/index.ts (hunk 3 above)
Comment (weeebr) [reply src/wallet/index.ts:305]: more modern way and more concise. like it!

**Comment by tgmarinho** [comment]
this conflict seems trick becareful

**Review by tgmarinho — APPROVED** [review]

**Pull request merged**